server/src/main/java/org/apache/iotdb/db/qp/physical/crud/QueryPlan.java

```diff
@@ -33,6 +35,12 @@
 
   private List<Path> paths = null;
   private List<TSDataType> dataTypes = null;
+
+  private List<Path> deduplicatedPaths = null;
+  private List<TSDataType> deduplicatedDataTypes = null;
+  private List<String> deduplicatedAggregations = new ArrayList<>();
+  private Map<Path, TSDataType> dataTypeMapping = new HashMap<>();
```

Comment:
This field may be useless, you can use deduplicatedPaths and deduplicatedDataTypes.